apps/cpp_rpc/CMakeLists.txt

```diff
@@ -32,6 +32,14 @@ if (OS)
    endif()
 endif()
 
+if(USE_OPENCL)
+   if (ANDROID_ABI)
+     if($ENV{ANDROID_NDK_MAJOR} VERSION_LESS "23")
+       set_property(TARGET tvm_rpc PROPERTY LINK_FLAGS -fuse-ld=gold)
```

Comment:
Also, I don't understand how `-fuse-ld=gold` helps to fix your [issue](https://github.com/apache/tvm/pull/13118#issuecomment-1304477305). In the log, I see that the linker cannot find `libOpenCL.so`. By manually specifying `-fuse-ld=gold` you just force compiler to use `ld.gold` and how changing linker helps you find this library? Unfortunately, I'm not able to reproduce this issue locally with docker. I don't have access to pull the `adreno_ci` docker image.

Reply:
Yes, ```-fuse-ld=gold``` helps to find the lib and link it properly.

Reply:
With NDK 23 and above and `ld.gold`, the host `ld` will be used. That's why we have the failure `ld.gold: --no-rosegment: unknown option` - host `ld` doesn't have this option. My guess is `ld.gold` in older NDK may be able to get a path to `libOpenCL.so` through host `ld`. I think in @srkreddy1238 's env, `libOpenCL.so` exists in the host, but not in NDK (I don't think any NDK comes with `libOpenCL.so`).

Reply:
Yes, for me, it's the same: `libOpenCL.so` is not a part of the NDK. When I build `tvm_rpc` with OpenCL support than I pass explicitly path to OpenCL SDK: `-DUSE_OPENCL=<path/to/sdk>` and it works fine also with `ld`. As I have mentioned in #13118, I checked that build works on many versions of NDK: 20.0.5594570, 21.3.6528147, 21.4.7075529, 22.1.7171670, etc. And I was surprised that it doesn't work in the docker.
I'm curious why, in case of @srkreddy1238 `ld` cannot find the `libOpenCL.so` but `ld.gold` can do it. In case then linkers have the same paths, I would expect that they will work in the same way.
Anyway, I don't think that this discussion is a stopper for merging this PR. But I'm just trying to understand.